homeassistant/components/recorder.py

```diff
@@ -168,11 +167,12 @@ def where_limit_to_run(self):
 class Recorder(threading.Thread):
     """A threaded recorder class."""
 
-    def __init__(self, hass):
+    def __init__(self, hass, config):
```

Comment:
The recorder should not be aware of a config existing. Instead, let's pass down the parameter for how many days to purge (or `None` if not to purge at all)